modules/restful_example/src/Plugin/resource/Articles__2_0.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful_example\Plugin\resource\Articles__1_0.
+ */
+
+namespace Drupal\restful_example\Plugin\resource;
+
+use Drupal\restful\Plugin\resource\ResourceEntity;
+use Drupal\restful\Plugin\resource\ResourceInterface;
+
+/**
+ * Class Articles
+ * @package Drupal\restful\Plugin\resource
+ *
+ * @Resource(
+ *   name = "articles:2.0",
+ *   resource = "articles",
+ *   label = "Articles",
+ *   description = "Export the article content type with cookie authentication.",
+ *   authenticationTypes = TRUE,
+ *   authenticationOptional = TRUE,
+ *   dataProvider = {
+ *     "entityType": "node",
+ *     "bundles": {
+ *       "article"
+ *     },
+ *     "viewMode": {
+ *       "name": "default",
+ *       "fieldMap": {
+ *         "body": "body",
+ *         "field_tags": "tags",
+ *         "field_image": "image",
+ *       }
+ *     }
+ *   },
+ *   majorVersion = 2,
+ *   minorVersion = 0
+ * )
+ */
+class Articles__2_0 extends ResourceEntity implements ResourceInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function publicFields() {
```

Comment:
Yes it is.

`Resource::publicFields` is abstract, and there is no default implementation in `ResourceEntity`. Maybe we need a follow up ticket to add a default implementation in `ResourceEntity`?